sdk/keyvault/keyvault-secrets/review/keyvault-secrets.api.md

```diff
@@ -64,6 +84,42 @@ export interface ProxyOptions {
     proxySettings?: string;
 }
 
+// @public
+export class RecoverDeletedSecretPoller extends Poller<RecoverDeletedSecretPollOperationState, SecretProperties> {
+    constructor(options: RecoverDeletedSecretPollerOptions);
+    delay(): Promise<void>;
+    intervalInMs: number;
+}
+
+// @public (undocumented)
+export interface RecoverDeletedSecretPollerOptions {
```

Comment:
It shouldn't be exported. I'll look into it!

Reply:
Here's the thing, the pollers classes needs to be exported, since that's what the begin methods return. However, should we export the entirety of the poller? since they're not supposed to be initialized outside of the client. cc: @jonathandturner & @xirzec

Reply:
Might be worth looking into how much the user needs to know about the poller if they can't change any of it.  You could split it into the "public" and "whole" poller, and then only give them the public side.

Reply:
That's weird in my head. Maybe if I return an interface that matches the poller's methods? without returning the actual poller?

Reply:
What about exposing this: https://github.com/Azure/azure-sdk-for-js/pull/5656 ?

Reply:
Interfaces are generally preferable to concrete classes, so seems like a good idea! :heart: